packages/dd-trace/src/appsec/reporter.js

```diff
@@ -171,6 +180,9 @@ function finishRequest (req, res) {
 
   incrementWafRequestsMetric(req)
 
+  // collect some headers even when no attack is detected
+  rootSpan.addTags(filterHeaders(req.headers, IDENTIFICATION_HEADERS_MAP))
```

Comment:
it's guaranteed `rootSpan` has always the `addTags` method? I'm asking because iast checks if it exists before calling it... [example](https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/appsec/iast/index.js#L70)

Reply:
why are we doing it?
In a quick search it looks like your example and other point are the only points where we are checking it...

Reply:
Yes, we check that `rootSpan` exists, and if it exists then it must have the `addTags` method. Everywhere in dd-trace code it assumes that at least. [example](https://github.com/search?q=repo%3ADataDog%2Fdd-trace-js+addTags%28+path%3A%2F%5Epackages%5C%2Fdd-trace%5C%2Fsrc%5C%2F%2F&type=code)

Reply:
What you should be checking is that the span exists, not that if setTags exists, the code you linked might actually crash because of that

Reply:
also if you're only using the rootSpan, then why are you saving the whole web.js context in the first place ? just do `web.root()` and you get the rootspan

Reply:
This is not discussion for this PR 😅 , but rootSpan is not the only property used in the code that Igor shared, topContext is also used.